### PR TITLE
[FEATURE] Add DKT Model

### PR DESCRIPTION
**Pull request by sone47**
Thanks for sending a pull request! 
Please make sure you click the link above to view the [contribution guidelines](../blob/master/CONTRIBUTE.md), 
then fill out the blanks below.

## Description ##
(Brief description on what this PR is about)

### What does this implement/fix? Explain your changes.
Add DKT model and relevant files.

#### Pull request type
- [ ] [DATASET] Add a new dataset
- [ ] [BUGFIX] Bugfix
- [x] [FEATURE] New feature (non-breaking change which adds functionality)
- [ ] [BREAKING] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] [STYLE] Code style update (formatting, renaming)
- [ ] [REFACTOR] Refactoring (no functional changes, no api changes)
- [ ] [BUILD] Build related changes
- [ ] [DOC] Documentation content changes
- [ ] [Sync] Synchronization with a repository 
- [ ] [OTHER] Other (please describe): 

#### Changes
- Add DKT model
- Add DKT examples
- Add KPT/EKPT doc
- Add tests for DKT

### Does this close any currently open issues?
N/A

### Any relevant logs, error output, etc?
N/A

## Checklist ##
Before you submit a pull request, please make sure you have to following:

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [FEATURE], [BREAKING], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage and al tests passing
- [x] Code is well-documented (extended the README / documentation, if necessary)
- [x] If this PR is your first one, add your name and github account to [AUTHORS.md](../blob/master/AUTHORS.md)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by tswsxk** [comment]
1. There are conflicting files. Please solve the conflicts before you merge.

File: pytest.ini [hunk 1]
```diff
@@ -13,5 +13,5 @@ flake8-max-line-length = 120
 # Others can be found in https://flake8.pycqa.org/en/latest/user/error-codes.html
 flake8-ignore = E402 F401 F403
 
-# --doctest-modules is used for unitest
-addopts = --doctest-modules --cov --cov-report=term-missing --flake8
+# --doctest-modules is used for unittest
```
Comment (tswsxk) [inline pytest.ini:16]: why modify these two lines?

**Review by tswsxk — COMMENTED** [review]
Please state the reasons for the  modification.

**Review by tswsxk — COMMENTED** [review]
Please state the reason that why the commented lines should be  changed

File: pytest.ini (hunk 1 above)
Comment (sone47) [reply pytest.ini:16]: 1. I think "unitest" is a spelling mistake for "unittest"
2. I don't know why only change "addopts" to "adopts" can I run pytest to test
Sorry. If these changes is not necessary, I will keep previous code of this file.

Comment (sone47) [reply pytest.ini:16]: I reviewed my code and ran it again. I think the first change could be kept, but the second change is my mistake.
I have committed again. I'm so sorry for that.

Comment (tswsxk) [reply pytest.ini:16]: I believe it should be ``unittes`` rather than ``unitest``

**Review by tswsxk — APPROVED** [review]

**Pull request merged**